kinto/core/utils.py

```diff
@@ -498,22 +498,48 @@ def parse_resource(resource):
     :param str resource: a uri formatted /buckets/<bid>/collections/<cid> or <bid>/<cid>.
     :returns: a dictionary with the bucket_id and collection_id of the resource
     """
-
     error_msg = 'Resources should be defined as '
-    "'/buckets/<bid>/collections/<cid>' or '<bid>/<cid>'. "
+    "'/buckets/<bid>/collections/<cid>' or '<bid>/<cid>' or "
+    "'/buckets/<bid>/collections/<cid>/records/<rid>.'"
     'with valid collection and bucket ids.'
-
+    resource_bucket = False
+    resource_record = False
     from kinto.views import NameGenerator
     id_generator = NameGenerator()
     parts = resource.split('/')
     if len(parts) == 2:
+        """<bid>/<cid>"""
         bucket, collection = parts
+    elif len(parts) == 3:
+        """
+        only bucket; </buckets/bid>
+        """
+        _, _, bucket = parts
+        resource_bucket = True
     elif len(parts) == 5:
         _, _, bucket, _, collection = parts
+    elif len(parts) == 7:
+        """for record"""
+        _, _, bucket, _, collection, _, record = parts
+        resource_record = True
     else:
         raise ValueError(error_msg)
-    if bucket == '' or collection == '':
-        raise ValueError(error_msg)
+    if resource_bucket:
+        if not id_generator.match(bucket):
+            raise ValueError(error_msg)
+        return {
+            'bucket': bucket
+        }
+    if resource_record:
+        if not id_generator.match(bucket) \
+                or not id_generator.match(collection) \
+                or not id_generator.match(record):
+            raise ValueError(error_msg)
+        return {
+            'bucket': bucket,
+            'collection': collection,
+            'record': record
+        }
```

Comment:
Oh, I didn't realize this was here! Why do we have both this and `view_lookup` (which is what I used when I wanted to parse URL-y things in kinto-changes)? I'm guessing based on your change that the event filtering stuff uses this code and not `view_lookup`. This function supports something that is sort of URL-y but also `<bid>/<cid>` which I think is kind of gross. Since we're introducing a new feature, I think it would be better if it could only parse URLs and not semi-URLs like this function does -- is it possible for this feature to use `view_lookup` instead of this? If it's difficult, let me know, maybe we can find another approach.